internal_transfers/actions/src/utils.ts

```diff
@@ -37,3 +37,20 @@ export async function getTxDataFromHash(
     logs,
   };
 }
+
+export function ethDeltaFromTraces(traces: Trace[]): Map<string, bigint> {
```

Comment:
Maybe make the types stronger here (ie return Map<address, bigint> maybe even reusing https://docs.ethers.org/v5/api/utils/address/). The issue I have with string is that the information that this has to be lower case is implicitly hidden in this method below (whereas if this was a strict type it could ensure lower casing when being created from a string), but the type is leaked outside of just internal utilities.

Reply:
Thanks! I have been asking my reviews for months now if this kinda thing existed. Finally got the answer I was hoping for and now I finally can go through and remove all that lower case casting.

Reply:
Also none of this seems to exist in [ethers v6](https://docs.ethers.org/v6/api/address/) (there is no Address type)